src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1041,6 +1041,8 @@ private void sendPing() {
 
         private InetSocketAddress rwServerAddress = null;
 
+        private InetSocketAddress serverAddress = null;
```

Comment:
This seems kinda bogus to me - why push this up to a field. Can't we determine the server address in "run" method, as a local variable, and call startConnect with that as an argument? That seems like an improvement to startConnect call at the same time. What do you think @anmolnar , does that make sense or am I missing something?

Reply:
Good idea. I've made the change.